etc/cmake/options.cmake

```
@@ -257,6 +257,11 @@ if(OT_MULTIPLE_INSTANCE)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_MULTIPLE_INSTANCE_ENABLE=1")
 endif()
 
+option(OT_PING_SENDER "enable ping sender support")
+if(OT_PING_SENDER OR OT_APP_CLI)
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_PING_SENDER_ENABLE=1")
+endif()
```

nit:
```suggestion
option(OT_PING_SENDER "enable ping sender support" ${OT_APP_CLI})
if(OT_PING_SENDER)
    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_PING_SENDER_ENABLE=1")
endif()
```

Per CMake documentation for [option](https://cmake.org/cmake/help/latest/command/option.html):
> option(<variable> "<help_text>" [value])
Provides an option for the user to select as ON or OFF. If no initial <value> is provided, OFF is used. If <variable> is already set as a normal or cache variable, then the command does nothing (see policy CMP0077).

The suggested change gives a chance to explicitly disable `OT_PING_SENDER` when `OT_APP_CLI` is enabled.

---

Good suggestion. Thanks @wgtdkp. Changed it in the last push.